storage/jellyfish_merkle/src/lib.rs

```diff
@@ -91,6 +92,15 @@ pub trait TreeReader {
 
     /// Gets node given a node key. Returns `None` if the node does not exist.
     fn get_node_option(&self, node_key: &NodeKey) -> Result<Option<Node>>;
+
+    /// Gets the rightmost leaf. Note that this assumes we are in the process of restoring the tree
+    /// and all nodes are at the same version.
```

Comment:
Lack of real version info in a restored state tree could make write amplification look really bad upon pruning, because we are touching the whole spectrum of all keys during pruning of any (old) version, which imposes compaction on almost all existing ssts. And the negative effect lasts longer with a larger version at which the backup was created (could mean virtually forever after several years of running.), which eliminates the gain from Jellyfish on a node restored from a backup. 

Seriously, shall we preserve the version info in a snapshot?